### PR TITLE
VIH-7415 participants with interpreters

### PR DESCRIPTION
**Pull request by phillipjchambers**
### JIRA link (if applicable) ###

VIH-7415

### Change description ###

* prevent participants with linked participants (currently interpreters and interpretees) from starting private consultations
* in the Start Private Consultation dialog: 
*  - group interpreter and interpretee together
*  - show participants as Unavailable if their linked participants are unavailable
* whilst in a consultation:
*  - group interpreter and interpretee together
*  - do not show witness interpreters in the participant list
*  use a trackBy with ngFor to prevent frequent repaints
* add/remove tests

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```

### CODE REVIEW

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/participant-waiting-room.component.ts [hunk 1]
```diff
@@ -239,7 +228,9 @@ export class ParticipantWaitingRoomComponent extends WaitingRoomBaseDirective im
     }
 
     get canStartJoinConsultation() {
-        return !this.isWitness && !this.isObserver && !this.isInterpreter && !this.isInterpretee;
+        return (
+            !this.isWitness && !this.isObserver && (!this.participant.linked_participants || !this.participant.linked_participants.length)
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/participant-waiting-room.component.ts:232]: in the base class, there is `isOrHasWitnessLink()` which you could re-use here

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/private-consultation-participants/private-consultation-participants.component.ts [hunk 2]
```diff
@@ -112,9 +118,18 @@ export class PrivateConsultationParticipantsComponent extends WRParticipantStatu
...
         } else {
-            return this.participantsInConsultation.filter(
-                p => p.hearing_role !== HearingRole.WITNESS && p.hearing_role !== HearingRole.OBSERVER
-            );
+            return this.participantsInConsultation
+                .filter(p => p.hearing_role !== HearingRole.WITNESS && p.hearing_role !== HearingRole.OBSERVER)
+                .filter(p => p.hearing_role !== HearingRole.INTERPRETER)
+                .map(p => {
+                    const interpreterLink = p.linked_participants.find(x => x.link_type === LinkType.Interpreter);
+                    // TODO (phill) type this properly
```
Comment (shaed-parkar) [inline VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/private-consultation-participants/private-consultation-participants.component.ts:126]: pending todo here

**Review by shaed-parkar — COMMENTED** [review]
few comments for you to check out

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/private-consultation-participants/private-consultation-participants.component.ts (hunk 2 above)
Comment (phillipjchambers) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/private-consultation-participants/private-consultation-participants.component.ts]: Already on it. You beat me to it 😆

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/participant-waiting-room.component.ts (hunk 1 above)
Comment (phillipjchambers) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/participant-waiting-room.component.ts:232]: I agree with you - makes sense to use it. Look though:
image
I'm gonna remove that func & just use the one in the base everywhere

**Review by valentinabennett — DISMISSED** [review]

File: VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/participant-waiting-room.component.ts (hunk 1 above)
Comment (shaed-parkar) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/participant-waiting-room.component.ts]: sorry which func do you plan to remove? the `isWitness` or `isWitnessOrHasWitnessLink`

Comment (phillipjchambers) [reply VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/participant-waiting-room/participant-waiting-room.component.ts]: isWitness

**Review by shaed-parkar — APPROVED** [review]

**Pull request merged**